### PR TITLE
chore(backend): update createTransfer options

### LINKED ISSUE #964: Do not fetch the original txn when doing a rollback or commit.
The debit/credit accounts may be set to `0n`, which will allow TB to lookup the original debit/credit accounts.

related to: 
- #968 
- #966

### LINKED ISSUE #966: Remove `post` and replace `pendingId` with optional `postId` and `voidId`.
You can make them mutually exclusive by removing `pendingId` renaming `CreateTransferOptions` to `CreateTransferOptionsBase `and adding something like 

```
export interface CommitTransferOptions extends CreateTransferOptionsBase {
  commitId?: string | bigint
  rollbackId?: never
}

export interface RollbackTransferOptions extends CreateTransferOptionsBase {
  commitId?: never
  rollbackId?: string | bigint
}

export type CreateTransferOptions = CommitTransferOptions | RollbackTransferOptions
```

To TB, `commitId` and `rollbackId` is the same (was combined into `pendingId`), so this can be consolidated.

Related to:
- #964 
- #968

### LINKED ISSUE #968: Set amount to `0n` for void/post
Setting amount to `0n` in transfer void/post in order for TigerBeetle to perform a lookup on the original pending transfer.
Simplifying the integration.

Related to 
- #964, 
- #966

### PR DESCRIPTION
**Pull request by wilsonianb**
## Changes proposed in this pull request

- 

## Context

- fixes #966 
- fixes #968 
- fixes #964 

## Checklist

- [x] Related issues linked using `fixes #number`
- [ ] Tests added/updated
- [ ] Documentation added
- [x] Make sure that all checks pass

### CODE REVIEW

File: packages/backend/src/accounting/transfers.ts [hunk 1]
```diff
@@ -16,57 +16,90 @@ type TransfersError = {
   error: TransferError
 }
 
-export interface CreateTransferOptions {
-  id?: string | bigint
+export interface NewTransferOptions {
```
Comment (sabineschaller) [inline packages/backend/src/accounting/transfers.ts:19]: Why "New"?

Comment (wilsonianb) [reply packages/backend/src/accounting/transfers.ts:19]: vs `post`ing or `void`ing an existing transfer

Comment (sabineschaller) [reply packages/backend/src/accounting/transfers.ts:19]: But why not keeping `CreateTransferOptions` or just `TransferOptions`? 

I don't have strong feelings about it but my first thought was "what is `OldTransferOptions`?" Maybe that's just me, though 😅

Comment (sabineschaller) [reply packages/backend/src/accounting/transfers.ts:19]: I think I get it now from reading your[ comment here](https://github.com/interledger/rafiki/issues/965#issuecomment-1380590708).

**Comment by sabineschaller** [comment]
Isn't this issue also related to your changes? 
- #965

**Comment by wilsonianb** [comment]
> Isn't this issue also related to your changes?
> 
>     * [Throw an error when an invalid `transfer.id` is provided for `accounting/transfers.ts#createTransfers()` #965](https://github.com/interledger/rafiki/issues/965)

I wasn't sure
- https://github.com/interledger/rafiki/issues/965#issuecomment-1398967422

**Review by sabineschaller — APPROVED** [review]
Looks good to me.

**Pull request merged**